agent_integration_test.go

```diff
@@ -0,0 +1,79 @@
+//go:build integration
+// +build integration
+
+package tfe
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAgentsRead(t *testing.T) {
+	//skipIfFreeOnly(t)
+
+	client := testClient(t)
+	ctx := context.Background()
+
+	org, orgCleanup := createOrganization(t, client)
+	t.Cleanup(orgCleanup)
+
+	upgradeOrganizationSubscription(t, client, org)
+
+	agentPool, agentPoolCleanup := createAgentPool(t, client, org)
+	t.Cleanup(agentPoolCleanup)
+
+	agent, agentCleanup := createAgent(t, client, org, agentPool)
```

Comment:
I think we can circumvent the agentPool reuse problem that @nfagerlund highlighted by always creating a new agentpool within createAgent and returning the pool, the agent, and the cleanup function.